app/presenters/tree_builder.rb

```
@@ -7,98 +7,11 @@ class TreeBuilder
-    when :schedules        then TreeBuilderReportSchedules
-    when :widgets          then TreeBuilderReportWidgets
-
-    # containers explorer tree
-    when :containers         then TreeBuilderContainers
-    when :containers_filter  then TreeBuilderContainersFilter
-
-    # automate explorer tree
-    when :ae               then TreeBuilderAeClass
-
-    # miq_ae_customization explorer trees
-    when :ab                    then TreeBuilderButtons
-    when :dialogs               then TreeBuilderServiceDialogs
-    when :dialog_import_export  then TreeBuilderAeCustomization
-    when :old_dialogs           then TreeBuilderProvisioningDialogs
-
-    # OPS explorer trees
-    when :diagnostics           then TreeBuilderOpsDiagnostics
-    when :rbac                  then TreeBuilderOpsRbac
-    when :servers_by_role       then TreeBuilderServersByRole
-    when :roles_by_server       then TreeBuilderRolesByServer
-    when :settings              then TreeBuilderOpsSettings
-    when :vmdb                  then TreeBuilderOpsVmdb
-
-    # PXE explorer trees
-    when :customization_templates then TreeBuilderPxeCustomizationTemplates
-    when :iso_datastores          then TreeBuilderIsoDatastores
-    when :pxe_image_types         then TreeBuilderPxeImageTypes
-    when :pxe_servers             then TreeBuilderPxeServers
-
-    # Services explorer tree
-    when :svcs                    then TreeBuilderServices
 
-    when :sa                      then TreeBuilderStorageAdapters
-
-    # Datastores explorer trees
-    when :storage     then TreeBuilderStorage
-    when :storage_pod then TreeBuilderStoragePod
-
-    when :datacenter              then TreeBuilderDatacenter
-    when :vat                     then TreeBuilderVat
-
-    when :network                 then TreeBuilderNetwork
-    when :df                      then TreeBuilderDefaultFilters
-    end
+  def self.class_for_type(type)
+    raise('Obsolete tree type.') if type == :filter
```

should we raise an error if `type` is not part of `X_TREE_NODE_CLASSES` hash?